designs/2024-per-rule-autofix-configuration/README.md

```diff
@@ -0,0 +1,406 @@
+(since the current implementation turns off suggestions when `disableFixes: false`, 
+but we need them if the fixer is disabled). 
+
+Inside `ReportTranslator` we can also convert the fix into a suggestion.
+
+##### Disabling for directives
+
+I think all the work in this regard can be done in the 
+[linter/apply-disabled-directives.js](https://github.com/eslint/eslint/blob/0f49329b4a7f91714f2cd1e9ce532d32202c47f4/lib/linter/apply-disable-directives.js#L466) 
+assuming the full entry of `reportUnusedDisableDirectives` configuration is available.
+
+It's unclear whether suggestions are currently supported for directives - please advise if they are, and where that logic resides.
+
+## Documentation
+
+<!--
+    How will this RFC be documented? Does it need a formal announcement
+    on the ESLint blog to explain the motivation?
+-->
+
+### Autofix on rules
+
+I think it is necessary to demonstrate the use of the full form of the record 
+in the [Configuration Files > Configuring rules](https://eslint.org/docs/latest/use/configure/configuration-files#configuring-rules) section,
+that way we show all the available options literally on one screen.
+
+I think there should be an explanatory section in the [Configure rules](https://eslint.org/docs/latest/use/configure/rules) section, 
+and maybe it should be the first one on the page to show all the available options at once so that the user is not intimidated 
+when they see multiple options on the same screen at the same time like:
+
+```js
+/* eslint eqeqeq: "off", curly: { severity: "error", autofix: false } */
+```
+
+I also believe that there should be a separate "Disabling autofix" section,
+probably located before the [Disabling rules](https://eslint.org/docs/latest/use/configure/rules#disabling-rules) 
+section, so that it can be directly referenced.
+
+### `linterOptions.reportUnusedDisableDirectives`
+
+A note should be added in the
+[Configuration Files > Reporting Unused Disable Directives](https://eslint.org/docs/latest/use/configure/configuration-files#reporting-unused-disable-directives)
+section that autofix behavior can be configured using the extended format.
+
+## Drawbacks
+
+<!--
+    Why should we *not* do this? Consider why adding this into ESLint
+    might not benefit the project or the community. Attempt to think 
+    about any opposing viewpoints that reviewers might bring up. 
+
+    Any change has potential downsides, including increased maintenance
+    burden, incompatibility with other tools, breaking existing user
+    experience, etc. Try to identify as many potential problems with
+    implementing this RFC as possible.
+-->
+
+The primary drawback is the complexity of normalizing configuration formats 
+across the codebase, which requires careful implementation across multiple packages (?).
+
+It's important to ensure nothing breaks during this transition.
+
+## Backwards Compatibility Analysis
```

Comment:
If [`eslint.calculateConfigForFile(filePath)`](https://eslint.org/docs/latest/integrate/nodejs-api#-eslintcalculateconfigforfilefilepath) would be returning rule configs normalized to the new object format, I think that could be considered a breaking change,